### DISCUSSION TITLE
Streaming conformer-transducer?

### DISCUSSION THREAD

**Discussion by maxeduc** [espnet/espnet#4697] [Q&A] [answered]
I'm wondering how to create a conformer-transducer in espnet (i.e. an RNN-T but with a conformer encoder instead of an RNN). AFAIK conformers are by default not-streaming unless they are trained with future self-attention context masked. [This documentation](https://espnet.github.io/espnet/tutorial.html#transducer) says that you can select "conformer" as the encoder for a transducer model in espnet, but I can't tell if this has future context masked. Something like this paper is what I'm talking about I think: https://arxiv.org/pdf/2011.10798.pdf

P.S. I may have no idea what I'm talking about, I just learned about how transformers work like yesterday.

**Comment by b-flo** [comment, marked as answer]
Hi,

We don't support streaming Transducer in ESPnet1, only ESPnet2. You can find the documentation [here](https://espnet.github.io/espnet/espnet2_tutorial.html#transducer-asr) which include a section for [streaming](https://espnet.github.io/espnet/espnet2_tutorial.html#streaming).

Also, we don't have a recipe example at the moment but I'll make a PR for Librispeech-100 later this day (both streaming and offline).

**Reply by maxeduc** [reply]
Thanks! Do you know if training a conformer encoder with the ESPnet2 Transducer will mask or ignore the right self-attention context so the model is totally causal?

**Reply by b-flo** [reply]
We don't deal with the right context during training (you can use an arbitrary value for that during decoding though) and causal convolution is used for the conformer's convolution module.

**Reply by maxeduc** [reply]
Got it, thanks again for the info!
